src/main/java/com/github/fppt/jedismock/operations/RO_scan.java

```diff
@@ -0,0 +1,64 @@
+package com.github.fppt.jedismock.operations;
+
+import com.github.fppt.jedismock.Utils;
+import com.github.fppt.jedismock.server.Response;
+import com.github.fppt.jedismock.server.Slice;
+import com.github.fppt.jedismock.storage.RedisBase;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.github.fppt.jedismock.Utils.convertToLong;
+
+class RO_scan extends AbstractRedisOperation {
+
+    private static final long CURSOR_START = 0;
+    // From the Redis documentation, the default count if not specified:
+    private static final long DEFAULT_COUNT = 10;
+
+    private static final String MATCH = "match";
+    private static final String COUNT = "count";
+
+    RO_scan(RedisBase base, List<Slice> params) {
+        super(base, params);
+    }
+
+    Slice response() {
+
+        Slice cursorSlice = params().get(0);
+        long cursor = cursorSlice != null ? convertToLong(cursorSlice.toString()) : CURSOR_START;
+
+        Slice matchSlice = extractParameter(params(), MATCH);
+        String match = matchSlice != null ? matchSlice.toString() : "*";
+
+        Slice countSlice = extractParameter(params(), COUNT);
+        long count = countSlice != null ? convertToLong(countSlice.toString()) : DEFAULT_COUNT;
+
+        String regex = Utils.createRegexFromGlob(match);
+        List<Slice> matchingKeys = base().keys().stream()
+                .skip(cursor)
+                .limit(count)
+                .filter(x -> x.toString().matches(regex))
+                .map(Response::bulkString)
+                .collect(Collectors.toList());
+
+        cursor = cursor + count;
+        if (cursor >= base().keys().size()) {
+            cursor = CURSOR_START;
+        }
+
+        List<Slice> response = Lists.newArrayList(Response.bulkString(Slice.create(String.valueOf(cursor))), Response.array(matchingKeys));
+        return Response.array(response);
+    }
+
+    private static Slice extractParameter(List<Slice> params, String name) {
```

Comment:
So I am being a bit nitpicky but I would say rather return an `Optional<Slice>` here. This will clean up your consumer calls, i.e. rather than:

```
Slice matchSlice = extractParameter(params(), MATCH);
String match = matchSlice != null ? matchSlice.toString() : "*";
```

to 

```
String match = extractParameter(params(), MATCH).orElse("*")
```

Reply:
Done! It's a bit more involved as `extractParameter` returns `Slice` and not `String`.